rds/rds_provider.go

```diff
@@ -251,6 +251,7 @@ func convertSpecToInput(v *crd.Database, subnetName string, securityGroups []str
 		MultiAZ:               aws.Bool(v.Spec.MultiAZ),
 		StorageEncrypted:      aws.Bool(v.Spec.StorageEncrypted),
 		BackupRetentionPeriod: aws.Int64(v.Spec.BackupRetentionPeriod),
+		DeletionProtection:    aws.Bool(v.Spec.DeleteProtection),
```

Comment:
will it apply for the existing RDS instances?

Reply:
no the rds-operator currently does not support changes of the crd.

Reply:
I guess we could add it??

Reply:
Let's do it in another PR though :)

Reply:
IMO creation and updates should run the same code, checking status of resources and reconciling. Like kubebuilder <3